Source/WebCore/page/EventHandler.cpp

```diff
@@ -4239,7 +4244,7 @@ bool EventHandler::handleDrag(const MouseEventWithHitTestResults& event, CheckDr
         return true;
     
     // Careful that the drag starting logic stays in sync with eventMayStartDrag().
-    if (m_mouseDownMayStartDrag && !dragState().source) {
+    if (m_mouseDownMayStartDrag && !draggedElement()) {
```

Comment:
I'm a little confused about why we don't want to hold a single `protectedDragElement()` during the scope of the `handleDrag()` method in its entirety, rather than repeatedly calling `draggedElement()`.

Does the result of `draggedElement()` change as we pass through this code?